doc/layer2-forwarding-enhancements/SONiC Layer 2 Forwarding Enhancements HLD.md

```diff
@@ -0,0 +1,444 @@
+
+
+Vlanmgr will handle Static FDB configuration from FDB table in CONFIG_DB. It will do validation and then set the static FDB in FDB_TABLE in APP_DB, which will be processed by FdbOrch.
+
+
+## 3.4 SyncD
+No change.
+
+
+## 3.5 SAI
+No changes are being made in SAI. The following SAI attributes are used by the changes being made to SONiC:
+
+1. SAI_FDB_FLUSH_ATTR_BRIDGE_PORT_ID
+2. SAI_FDB_FLUSH_ATTR_BV_ID
+3. SAI_FDB_FLUSH_ATTR_ENTRY_TYPE
+4. SAI_FDB_ENTRY_ATTR_TYPE
+5. SA_FDB_EVENT_MOVE
+6. SAI_SWITCH_ATTR_FDB_AGING_TIME
+
+
+
+## 3.6 CLI
+
+### 3.6.1 Configuration Commands
+**FDB Aging time configuration**
+
+root@sonic:/# config mac aging-time <value>
+- To set the mac aging time to a value.
+
+
+**Static MAC configuration**
+
+
+root@sonic:/# config mac add 00:10:3a:2b:05:67 100 Ethernet2
+- To add a static mac on vlan 100 and port Ethernet2.
+
+
+root@sonic:/# config mac del 00:10:3a:2b:05:67 100
+- To delete a static mac on vlan 100.
+
+**VLAN Range configuration**
+
+root@sonic:/# config vlan range add <-w, optional argument>
+
+root@sonic:/# config vlan range del <-w, optional argument>
+
+root@sonic:/# config vlan member range add <interface_id> <-w, optional argument>
+
+root@sonic:/# config vlan member range del <interface_id> <-w, optional argument>
+
+
+
+### 3.6.2 Show Commands
+
+
+root@sonic:/# show mac aging-time
+- To display the current configured mac aging time.
+
+
+# 4 Flow Diagrams
+Flow diagrams are provided in the [Design](#3-design) section.
+
+# 5 Serviceability and Debug
+Debug counters will be added for all operations and events related to L2 forwarding like the following:
+In Orchagent:
+- Number of FDB learn/aged events received from ASIC_DB.
+- Number of FDB add/delete request received from APP_DB.
+- Number vlan add/delete received from APP_DB.
+- Number of FDB entries inserted in the saved FDB database.
+- Number of FDB add/delete requests sent to syncd.
+- Counters for various failure conditions.
+
+
+NOTICE/INFO level logs will be added for major operations and events in vlanmgr and orchagent.
+
+
+Logic is added in the functions for VLAN range commands to append warning messages to a list and are displayed if the optional argument is provided. Iteration in the loops is not intentionally broken, if an existing/non-existing VLAN is trying to be operated upon, to ensure configuration for other VLANs goes through successfully.
+
+# 6 Warm Boot Support
+
+No change.
+
+# 7 Scalability
+
+If the number of VLANs in the range commands is high, the time it takes to perform the operation increases, though it is significantly lower compared to the same operation being performed using existing commands.
+
+
+# 8 Unit Test
+
+**Data structure changes**
+
+1. Verify that FDB internal map is updated properly when a FDB entry is added/deleted in hardware or when a static FDB entry is added/deleted.
```

Comment:
what's the plan to perform the hardware check in the test? different ASIC vendors have different ways to check the hardware FDB entries.

Reply:
pytest will only check the ASIC_DB and not hardware. We usually verify hardware state by running traffic.